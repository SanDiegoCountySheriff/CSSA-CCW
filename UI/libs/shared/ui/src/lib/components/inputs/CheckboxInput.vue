<template>
  <v-checkbox
    v-model="value"
    :label="$t(label)"
    @change="handleChange"
  />
</template>

<script setup lang="ts">
import { ref } from 'vue';

export interface CheckboxInputProps {
  label: string;
  target: string;
}

const props = withDefaults(defineProps<CheckboxInputProps>(), {
  label: '',
  target: '',
});

const emit = defineEmits(['input']);
const value = ref(false);

function handleChange() {
  emit('input', value.value, props.target);
}
</script>
