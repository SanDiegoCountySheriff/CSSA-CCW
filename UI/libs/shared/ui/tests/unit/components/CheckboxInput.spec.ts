import Vue from 'vue';
import CheckboxInput from '@shared-ui/components/inputs/CheckboxInput.vue';
import Vuetify from 'vuetify';
import { createLocalVue, shallowMount } from '@vue/test-utils';

Vue.use(Vuetify);
const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('CheckboxInput', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    //@ts-ignore
    wrapper = shallowMount(CheckboxInput, {
      localVue,
      vuetify,
      mocks: tMock,
      propsData: {
        label: 'test',
        target: 'testTarget',
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('Should render the Checkbox', () => {
    expect(wrapper.props().label).toEqual('test');
    expect(wrapper.props().target).toEqual('testTarget');
  });

  it('Should emit the correct information', async () => {
    await wrapper.vm.$nextTick(() => {
      wrapper.vm.handleChange();
    });
    // false is the default value for the component.
    expect(wrapper.emitted().input[0][0]).toEqual('false');
    expect(wrapper.emitted().input[0][1]).toEqual('testTarget');
  });
});
