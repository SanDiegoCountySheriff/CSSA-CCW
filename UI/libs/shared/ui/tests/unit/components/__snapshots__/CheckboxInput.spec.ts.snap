// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxInput should match the snapshot 1`] = `
<div class="v-input theme--light v-input--selection-controls v-input--checkbox">
  <div class="v-input__control">
    <div class="v-input__slot">
      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-2" role="checkbox" type="checkbox" value="">
        <div class="v-input--selection-controls__ripple"></div>
      </div><label for="input-2" class="v-label theme--light" style="left: 0px; position: relative;">test</label>
    </div>
    <div class="v-messages theme--light">
      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
    </div>
  </div>
</div>
`;
